gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -881,7 +881,7 @@ public void nextNull() throws IOException {
    * @throws NumberFormatException if the next literal value cannot be parsed
    *     as a double, or is non-finite.
    */
-  public double nextDouble() throws IOException {
+  public Number nextNumber() throws IOException {
```

Comment:
This change breaks backward compatibility. Maybe it would be better to add `nextNumber()` as separate method.

Reply:
I agree with you that maybe nextdouble() should be marked @Derected